src/extended_json.ts

```diff
@@ -273,7 +273,10 @@ const BSON_TYPE_MAPPINGS = {
   MaxKey: () => new MaxKey(),
   MinKey: () => new MinKey(),
   ObjectID: (o: ObjectId) => new ObjectId(o),
-  ObjectId: (o: ObjectId) => new ObjectId(o), // support 4.0.0/4.0.1 before _bsontype was reverted back to ObjectID
+  // The _bsontype for ObjectId is spelled with a capital "D", to the mapping above will be used (most of the time)
```

Comment:
Why can we not just eradicate the capital "D" completely in the _bsontype? It's a runtime value - is this because we think that users may be storing the properties of the ObjectId and we need to be backwards compatible? (I'm thinking Compass preferences as an example).